src/components/ToDoHeader.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { FormEvent, useContext, useRef } from 'react';
+import { DispatchContext, StateContext } from './StateContext';
+import classNames from 'classnames';
+
+export const ToDoHeader = () => {
+  const { toDoTitle, todos, focusOnTodo } = useContext(StateContext);
+  const dispatch = useContext(DispatchContext);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const handleSubmit = (event: FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const trimmedToDoTitle = toDoTitle.trim();
+
+    if (!trimmedToDoTitle) {
+      return;
+    } else {
+      dispatch({
+        type: 'ADD_TODO',
+      });
+      dispatch({
+        type: 'WRITE_NEW_TITLE',
+        newTitle: '',
+      });
+    }
+  };
+
+  if (focusOnTodo) {
+    inputRef.current?.focus();
+  } else {
+    inputRef.current?.blur();
```

Comment:
I would just add useEffect in this component and on every new todo focus input.

Reply:
I added useEffect but I dont know what should I put in condition instead of focusOnTodo.

Reply:
in dependency array put todos so every time that they change you will focus on input

Reply:
When I use useEffect here, test with double clicking cannot pass.